test/integration/common/StudioActions.kt

```diff
@@ -105,26 +99,29 @@ object Utils {
         beforeRetry: (() -> Unit) = {},
         successCondition: () -> Boolean
     ) {
+        var success = false
         val deadline = System.currentTimeMillis() + 10_000
-        while (!successCondition() && System.currentTimeMillis() < deadline) {
-            beforeRetry()
-            delayAndRecompose(context, 500)
+        while (!success && System.currentTimeMillis() < deadline) {
+            try {
```

Comment:
Success conditions can throw exceptions in some cases! We see an example of this within `StudioActions`
```kt
StudioState.notification.queue.last().code == Message.Connection.TRANSACTION_COMMIT_SUCCESSFULLY.code()
```
What if there isn't anything in the queue at all? `.last()` throws an error - we don't want test writers to have to include try catch logic within their `successCondition` as the lambda becomes much too verbose.